optimum/exporters/onnx/__main__.py

```diff
@@ -74,6 +74,20 @@ def main():
         task, args.model, framework=args.framework, cache_dir=args.cache_dir, trust_remote_code=args.trust_remote_code
     )
 
+    if task + "-with-past" in TasksManager.get_supported_tasks_for_model_type(
+        model.config.model_type.replace("_", "-"), "onnx"
+    ):
+        if args.task == "auto":  # Make -with-past the default if --task was not explicitely specified
+            task = task + "-with-past"
+        elif args.task in ["seq2seq-lm", "causal-lm", "speech2seq-lm", "vision2seq-lm"]:
```

Comment:
I think we can infer task that can have a `with-past` flavor dynamically here.

Reply:
fixed thanks